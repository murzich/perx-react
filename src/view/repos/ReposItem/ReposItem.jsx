import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';
import TagsList from '../TagList';

const ReposItem = ({
  url,
  name,
  description,
  stars,
  watchers,
  forks,
  language,
}) => (
  <List.Item>
    <List.Content>
      <List.Header
        as="a"
        href={url}
      >
        {name}
      </List.Header>
      <List.Description>
        {description}
      </List.Description>


      <TagsList
        tags={{
          stars,
          forks,
          watchers,
          language,
        }}
      />
    </List.Content>
  </List.Item>
);

ReposItem.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  stars: PropTypes.number.isRequired,
  watchers: PropTypes.number.isRequired,
  forks: PropTypes.number.isRequired,
  language: PropTypes.string,
};

export default ReposItem;
