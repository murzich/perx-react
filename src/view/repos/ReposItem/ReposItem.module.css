.tags {
  margin-top: 1rem;
}
