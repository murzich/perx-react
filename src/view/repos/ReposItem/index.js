import ReposItem from './ReposItem';

export default ReposItem;
