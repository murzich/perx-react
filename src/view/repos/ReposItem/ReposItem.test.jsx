import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReposItem from './ReposItem';

describe('ReposItem', () => {
  afterEach(cleanup);

  it('should match snapshot without props', () => {
    const { asFragment } = render(<ReposItem />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot without props', () => {
    const props = {
      url: 'testUrl',
      name: 'testName',
      description: 'testDescription',
      stars: 42,
      watchers: 13,
      forks: 8,
      language: 'testLanguage',
    };
    const { asFragment } = render(<ReposItem {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
