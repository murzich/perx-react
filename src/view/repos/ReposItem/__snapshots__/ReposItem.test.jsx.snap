// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposItem should match snapshot without props 1`] = `
<DocumentFragment>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="content"
    >
      <a
        class="header"
      />
      <div
        class="description"
      />
      <div
        class="ui small labels tags"
      >
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
          0
        </div>
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="fork icon"
          />
          0
        </div>
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="eye icon"
          />
          0
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReposItem should match snapshot without props 2`] = `
<DocumentFragment>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="content"
    >
      <a
        class="header"
        href="testUrl"
      >
        testName
      </a>
      <div
        class="description"
      >
        testDescription
      </div>
      <div
        class="ui small labels tags"
      >
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
          42
        </div>
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="fork icon"
          />
          8
        </div>
        <div
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="eye icon"
          />
          13
        </div>
        <div
          class="ui label"
        >
          testLanguage
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
